Core/Object Arts/Dolphin/Base/Locale.cls

```diff
@@ -429,29 +429,34 @@ pmDesignator
 	^self subclassResponsibility!
 
 print: anObject on: aWriteStream format: formatString alphabet: formatChars formatters: aLookupTable
+	
 	| ch formatStream tokenStream |
 	tokenStream := formatString class writeStream: 9.
 	formatStream := formatString readStream.
 	ch := formatStream nextAvailable.
 	[ch isNil] whileFalse: 
-			[(formatChars includes: ch)
-				ifTrue: 
-					[| start |
-					tokenStream reset.
-					start := ch.
+		[ch == $'
+			ifTrue: 
+				[aWriteStream nextPutAll: (formatStream upTo: $').
+				ch := formatStream nextAvailable].
+		(formatChars includes: ch)
```

Comment:
There is a bug here. This change does need a regression test. I would suggest modifying the existing `LocaleTest>>testPrintDateTimeOnFormat` as follows:

```
testPrintDateTimeOnFormat
	| subject now expected |
	now := DateAndTime
				year: 2019
				day: 97
				hour: 18
				minute: 26
				second: 21 + (827 / 1000).
	subject := self defaultSubject.
	expected := '6:26 ', subject pmDesignator, ' (and 21.827 seconds) on Sunday, 7 of April 2019 somewhere'.
	{now. now asParameter. TimeStamp fromDateAndTime: now} do: 
			[:each |
			| actual |
			actual := String streamContents: 
							[:s |
							subject
								printDateTime: each
								on: s
								format: 'h:mm tt (an\d ssffffff \secon\d\s) on dddd, d ''of'' MMMM yyyy ''somewhere'''].
			self assert: actual equals: expected]
```

By having one of the quoted parts of the format string at the end, it will show up the bug.